GxVAEs : Two Joint VAEs Generate Hit Molecules from Gene Expression Profiles . The de novo generation of hit-like molecules that show bioactivity and drug-likeness is an important task in computer-aided drug discovery . Although artificial intelligence can generate molecules with desired chemical properties , most previous studies have ignored the influence of disease-related cellular environments . This study proposes a novel deep generative model called GxVAEs to generate hit-like molecules from gene expression profiles by leveraging two joint variational autoencoders ( VAEs ) . The first VAE , ProfileVAE , extracts latent features from gene expression profiles . The extracted features serve as the conditions that guide the second VAE , which is called MolVAE , in generating hit-like molecules . GxVAEs bridge the gap between molecular generation and the cellular environment in a biological system , and produce molecules that are biologically meaningful in the context of specific diseases . Experiments and case studies on the generation of therapeutic molecules show that GxVAEs outperforms current state-of-the-art baselines and yield hit-like molecules with potential bioactivity and drug-like properties . Hit identification , which involves discovering hit molecules with the desired bioactivity and therapeutic effects within an infinite chemical space , is a critical challenge in drug discovery ( Dobson et al . 2004 ) . Traditionally , hit identification campaigns have been conducted using experimental approaches such as high-throughput screening ( HTS ) ( Hertzberg and Pope 2000 ) . HTS provides a valuable methodology for identifying hits with desired bioactivity ( Scannell et al . 2022 ) . However , the hit rate observed in HTS campaigns tends to be relatively low , which results in many molecules that are inactive or that hold less promise being screened out ( Rahman et al . 2022 ) . Additionally , experimental approaches to drug discovery typically rely on laborintensive and time-consuming processes . De novo molecular generation using artificial intelligence ( AI ) techniques has emerged as a promising approach for computer-aided drug discovery . The power of deep learning and computational chemistry enables the generation of new molecules with desired bioactivity for specific therapeutic targets . Deep generative models , such as generative adversarial networks ( GANs ) ( Guimaraes et al . 2017 ; De Cao and Kipf 2018 ; Li et al . 2022 ) and variational autoencoders ( VAEs ) ( Oliveira , Da Silva , and Quiles 2022 ; Dollar et al . 2021 ; Kusner , Paige , and Hernández-Lobato 2017 ; Jin , Barzilay , and Jaakkola 2018 ) accelerate the drug discovery process by employing computational models to generate molecules having a specific bioactivity . By capturing the essential features of molecules , deep generative models can produce new molecules with similar structures and related properties . Most AI-driven molecular generation models focus on the generation of new molecules with improved chemical properties of interest . While omics-based approaches to drug discovery offer promising opportunities , it is essential to understand their limitations ( Kang , Ko , and Mersha 2022 ) . Interpretation of omics data in the context of drug discovery also requires a deep understanding of the underlying biological and disease mechanisms . In this study , we proposed GxVAEs to generate hit-like molecules from gene expression profiles using two joint VAEs ( i.e . , ProfileVAE and MolVAE ) . First , ProfileVAE functions as a feature extractor to extract latent features from gene expression profiles . The extracted features are then used as conditions to guide MolVAE in generating hit-like molecules . GxVAEs bridges the gap between molecular generation and the cellular environment of a biological system , making the produced molecules more biologically meaningful in the context of specific diseases . The main contributions of this study are summarized as follows : • Hit-like molecular generation that considered the cellular environment : This study considers the influence of the cellular environment on specific diseases during the molecular generation process . • A simple but effective model architecture : The combination of two VAEs successfully generated hit-like molecules from gene expression profiles . • Superior performance over state-of-the-art ( SOTA ) models : Experiments and case studies demonstrate that the proposed GxVAEs outperform the current SOTA models for the same objectives . De novo molecular generation aims to produce new molecules with specific chemical properties . Generally , the data structures used for de novo molecular generation are molecular graphs ( Manolopoulos and Fowler 1992 ) and simplified molecular input line entry system ( SMILES ) strings ( Weininger 1988 ) . TransVAE ( Dollar et al . 2021 ) and Gram-marVAE ( Kusner , Paige , and Hernández-Lobato 2017 ) are two deep generative models that produce molecules from SMILES strings using VAEs . JTVAE ( Jin , Barzilay , and Jaakkola 2018 ) uses a node tree to generate molecules from a molecular graph . It first creates a tree-structured scaffold on a substructure and then combines the tree-structured scaffold into a molecule using a graphical message-passing network ( Dai , Dai , and Song 2016 ) . MolGAN ( De Cao and Kipf 2018 ) and TransORGAN ( Li et al . 2022 ) are discrete GANs that use reinforcement learning to generate molecular graphs and SMILES strings , respectively . With the development of omics data analysis , especially gene expression profiles used in drug discovery ( Turanli et al . 2018 ; Chen et al . 2020 ) , chemists have begun to use comprehensive biological response information to generate therapeutic molecules for the treatment of specific diseases . ExpressionGAN ( Méndez-Lucio et al . 2020 ) and TRI-OMPHE ( Kaitoh and Yamanishi 2021 ) are the SOTA deep generative models that are most relevant to our study on generating hit-like molecules for target proteins with no prior annotation of the target training molecules . Expres-sionGAN is a GAN model that bridges systems biology and molecular design to produce molecules with a high probability of inducing the desired transcriptome features automatically from gene expression profiles . TRIOMPHE first calculates the correlation of ligand-target interactions between the chemically induced transcriptome profiles of cellular responses to molecular treatment and the transcriptome profiles of gene perturbations that reflect cellular responses to gene knockdown or gene overexpression of target proteins . Next , a VAE was employed to generate a new molecule with the desired transcriptome profile . However , ExpressionGAN generates hit-like molecules with low validity ( > 8.5% ) , and its ability to reproduce known ligands is limited . Additionally , TRIOMPHE only utilizes gene expression profiles in correlation calculations , whose VAE is not involved in molecular generation . To address these issues , GxVAEs were proposed in this study . By bridging the gap between the cellular environment and the generation of ther - The Thirty-Eighth AAAI Conference on Artificial Intelligence ( AAAI - 24 ) apeutic molecules , GxVAEs generate molecules that are biologically meaningful in the context of specific diseases using gene expression profiles . Figure 1 shows an overview of the GxVAEs . GxVAEs mainly consists of two joint VAEs , namely ProfileVAE and MolVAE . ProfileVAE acts as a feature extractor to extract low-dimensional feature vectors from high-dimensional gene expression profiles . MolVAE consists of bidirectional GRUs that are conditioned on the features of gene expression profiles using non-canonical SMILES ( i.e . , variant SMILES ( Li and Yamanishi 2023 ) ) strings to generate hitlike molecules . Note that teacher forcing ( Yan et al . 2023 ) is used to stabilize the training and accelerate the convergence of the GxVAEs during the training phase of the MolVAE . ProfileVAE . The goal of Profil-eVAE is to learn the marginal likelihood of gene expression profiles during the following generative process : max EQUATION where E [ • ] is an expectation operation , C is the latent variable , θ and ϕ denote the parameters of the ProfileVAE encoder and ProfileVAE decoder , respectively , and p ϕ ( X | C ) and q θ ( C | X ) are the likelihood function and posterior distribution , respectively . The loss function of the ProfileVAE can be formulated as follows : L G ( θ , ϕ , X , C , β ) = E q θ ( C | X ) [ log p ϕ ( X | C ) ] ( 2 ) - β • D KL ( q θ ( C | X ) | | p ϕ ( C ) ) , where β denotes the weight of the KL divergence D KL ( Joyce 2011 ) . MolVAE . Then , EQUATION ) Note that the feature vectors C of the extracted gene expression profiles are used as the conditions for MolVAE . MolVAE aims to maximize the lower bound of the true logmarginal likelihood as follows : EQUATION - D KL ( q ψ ( Z | S ) | | p φ ( Z ) ) . Three types of gene expression profiles were used to validate the effectiveness of the proposed GxVAEs . • Chemically induced profiles were collected from the LINCS L1000 database ( Duan et al . 2014 ) . Here , gene expression profiles of MCF7 cell line treated with 13 , 755 molecules were used . • Target perturbation profiles were obtained from the LINCS database . For example , we analyzed RAC-α serine / threonine-protein kinase ( AKT1 ) , RACβ serine / threonine-protein kinase ( AKT2 ) , Aurora B kinase ( AURKB ) , cysteine synthase A ( CTSK ) , epidermal growth factor receptor ( EGFR ) , histone deacetylase 1 ( HDAC1 ) , mammalian target of rapamycin ( MTOR ) , phosphatidylinositol 3 - kinase catalytic subunit ( PIK3CA ) , decapentaplegic homolog 3 ( SMAD3 ) , and tumor protein p53 ( TP53 ) , which are therapeutic proteins for cancers . • Disease-specific profiles were collected from the crowd extracted expression of differential signatures ( CREEDS ) database ( Wang et al . 2016 ) . Validity refers to the ratio of chemically valid molecules generated , which can be verified in practice using the RDKit tool ( Landrum 2013 ) . A quantitative estimate of drug-likeness ( QED ) quantifies the likelihood that a molecule is a drug ( Appendix D ) . To demonstrate that ProfileVAE has the ability to extract the biological features of gene expression profiles , we compared the distributions of the input gene expression profiles and the reconstructed profiles . The distributions of the reconstructed profiles approximate those of the corresponding validation data , which indicates that ProfileVAE extracted the features of the gene expression profiles well , and proved the effectiveness of the proposed GxVAEs . to those in the validation set . Furthermore , Table 1 lists the statistics for the validation set and molecules generated by the proposed GxVAEs . The average , maximum , minimum lengths , and the average molecular weight of the generated SMILES strings were basically consistent with the SMILES strings in the original validation set , indicating that GxVAEs learned the data distributions of the SMILES strings well . Table 2 shows the ability of the proposed Gx-VAEs to generate candidate molecules . Note that Expres-sionGAN has limited ability to generate valid molecules , thus we only compared GxVAEs to the TRIOMPHE baseline . The experimental results demonstrate that the validity of the proposed GxVAEs is three times higher than that of TRIOMPHE for generating ligand-like molecules using the gene expression profiles of the ten target proteins , reaching at least 78.0% ( HDAC1 ) . Moreover , the uniqueness of the GxVAEs exceeds that of TRIOMPHE except for PIK3CA ( 93.5% < 97.2% ) . Additionally , the novelty of the GxVAEs ( 97.7% ) is close to that of TRIOMPHE . Overall , the proposed GxVAEs provided a sufficient number of candidate ligands for the ten target proteins . Table 3 compares the Tanimoto coefficients of GxVAEs with those of the two SOTA models . The experimental results illustrate that the Tanimoto coefficients of the candidate ligands generated by GxVAEs for the ten target proteins were much higher than those of the two SOTA models . For example , the Tanimoto coefficient of GxVAEs for AKT1 was 2.7 and 2.0 times higher than those of the two baselines . The chemical structures of the hit-like molecules generated by GxVAEs are all similar to the structures of the known ligands . Overall , the proposed GxVAEs show excellent performance in generating hit-like molecules from the gene expression profiles , and the biological activity of the generated molecules far exceeded the SOTA baselines . Figure 5 shows the process undertaken by GxVAEs for therapeutic molecular generation . Finally , the disease reversal profiles were input into GxVAEs to generate candidate therapeutic molecules . We compared the results generated by GxVAEs with those generated by DRAGONET ( Yamanaka et al . 2023 ) . For a fair comparison , we used the same datasets on patient gene expression profiles and molecule chemical structures in DRAGONET . Figure 6 shows the therapeutic molecules generated by DRAGONET and GxVAEs , as well as the Tanimoto coefficients calculated relative to known approved drugs . The Tanimoto coefficients between hydrocortisone and the molecules generated by GxVAEs using the disease reversal profile of patients having atopic dermatitis reached 1.0 . This result indicates that GxVAEs effectively captured the structural features of the drugs approved for the treatment of atopic dermatitis . Furthermore , the molecules generated by GxVAEs for treating gastric cancer and Alzheimer's disease showed structural features that were similar to those of known approved drugs . Thus , the proposed GxVAEs can generate molecules that have higher therapeutic properties than DRAGONET . We proposed GxVAEs , which consisted of two joint VAEs ( i.e . , ProfileVAE and MolVAE ) , to generate hit-like molecules from gene expression profiles . ProfileVAE extracted the features of the gene expression profiles , which were then used as conditions to guide MolVAE in producing hit-like molecules . The experimental results showed that GxVAEs outperformed the current SOTA baselines and efficiently generated hit-like molecules from gene expression profiles . Furthermore , we showed the capability of GxVAEs to create molecular structures with the potential therapeutic effects for various diseases from patients ' disease profiles . One limitation of GxVAEs is that the diversity may be influenced by the size of the latent space . If an implementation of MolVAE adopts a fixed latent vector , it could potentially constrain the diversity of newly generated molecules .